<?xml version="1.0" encoding="UTF-8"?>
<pd:ProcessDefinition xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:ns="http://www.tibco.com/pe/EngineTypes" xmlns:pfx="http://www.tibco.com/pe/GenerateErrorActivity/InputSchema" xmlns:pd="http://xmlns.tibco.com/bw/process/2003" xmlns:ns1="http://www.tibco.com/pe/DeployedVarsType" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:pfx5="http://www.tibco.com/CommonLE2/namespace/private/Common" xmlns:tib="http://www.tibco.com/bw/xslt/custom-functions" xmlns:pfx4="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Category" xmlns:pfx3="http://www.tibco.com/CommonLE/namespace/Custom" xmlns:pfx2="http://xmlns.example.com/unique/default/namespace/1098735544074" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:pfx6="http://www.tibco.com/CommonLE2/Public/Common/Dummy">
    <xsd:import namespace="http://www.tibco.com/CommonLE/namespace/Custom" schemaLocation="/Shared Resources/Schema Definitions/Custom/ExceptionCategory.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/namespace/Public/Pluggable/GUI/Configuration/Category" schemaLocation="/SharedResources/SchemaDefinitions/Public/Pluggable/GUI/Administration/Configuration/ExceptionCategory.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/namespace/private/Common" schemaLocation="/SharedResources/SchemaDefinitions/Private/Common/LocalExceptions.xsd"/>
    <xsd:import namespace="http://www.tibco.com/CommonLE2/Public/Common/Dummy" schemaLocation="/SharedResources/SchemaDefinitions/Public/Pluggable/GUI/Common/DummyReturnMessage.xsd"/>
    <pd:name>Administration/DataAccess/Category/MSSQL/UpdateCategory.process</pd:name>
    <pd:description>Updates an existing exception category</pd:description>
    <pd:startName>Start</pd:startName>
    <pd:startType ref="pfx4:ExceptionCategories"/>
    <pd:startX>27</pd:startX>
    <pd:startY>101</pd:startY>
    <pd:returnBindings>
        <pfx6:Dummy>
            <pfx6:StatusMessage>
                <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/ReservedIDs/DummyID"/>
            </pfx6:StatusMessage>
        </pfx6:Dummy>
    </pd:returnBindings>
    <pd:endName>End</pd:endName>
    <pd:endType ref="pfx6:Dummy"/>
    <pd:endX>332</pd:endX>
    <pd:endY>102</pd:endY>
    <pd:errorSchemas>
        <Schema0 ref="pfx5:DBRecordNotFound"/>
    </pd:errorSchemas>
    <pd:processVariables/>
    <pd:targetNamespace>http://xmlns.example.com/1295662719142</pd:targetNamespace>
    <pd:activity name="Update Exception Category">
        <pd:type>com.tibco.plugin.jdbc.JDBCUpdateActivity</pd:type>
        <pd:resourceType>ae.activities.JDBCUpdateActivity</pd:resourceType>
        <pd:x>167</pd:x>
        <pd:y>110</pd:y>
        <config>
            <timeout>%%Database/MIG_Timeout%%</timeout>
            <commit>false</commit>
            <emptyStrAsNil>false</emptyStrAsNil>
            <jdbcSharedConfig>/SharedResources/Connections/Database/DBServerMSSQL.sharedjdbc</jdbcSharedConfig>
            <statement>Update ExceptionCategory set name =?, Description =?
Where ID = ?</statement>
            <Prepared_Param_DataType>
                <parameter>
                    <parameterName>CategoryName</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
                <parameter>
                    <parameterName>Description</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
                <parameter>
                    <parameterName>CategoryID</parameterName>
                    <dataType>VARCHAR</dataType>
                </parameter>
            </Prepared_Param_DataType>
        </config>
        <pd:inputBindings>
            <jdbcUpdateActivityInput>
                <CategoryName>
                    <xsl:choose>
                        <xsl:when test="exists($Start/pfx4:ExceptionCategories/pfx4:ExceptionCategory[1]/pfx4:CategoryName)">
                            <xsl:value-of select="$Start/pfx4:ExceptionCategories/pfx4:ExceptionCategory[1]/pfx4:CategoryName"/>
                        </xsl:when>
                        <xsl:otherwise>
                            <xsl:attribute name="xsi:nil">true</xsl:attribute>
                        </xsl:otherwise>
                    </xsl:choose>
                </CategoryName>
                <Description>
                    <xsl:choose>
                        <xsl:when test="exists($Start/pfx4:ExceptionCategories/pfx4:ExceptionCategory[1]/pfx4:Description)">
                            <xsl:value-of select="$Start/pfx4:ExceptionCategories/pfx4:ExceptionCategory[1]/pfx4:Description"/>
                        </xsl:when>
                        <xsl:otherwise>
                            <xsl:attribute name="xsi:nil">true</xsl:attribute>
                        </xsl:otherwise>
                    </xsl:choose>
                </Description>
                <CategoryID>
                    <xsl:choose>
                        <xsl:when test="exists($Start/pfx4:ExceptionCategories/pfx4:ExceptionCategory[1]/pfx4:CategoryID)">
                            <xsl:value-of select="$Start/pfx4:ExceptionCategories/pfx4:ExceptionCategory[1]/pfx4:CategoryID"/>
                        </xsl:when>
                        <xsl:otherwise>
                            <xsl:attribute name="xsi:nil">true</xsl:attribute>
                        </xsl:otherwise>
                    </xsl:choose>
                </CategoryID>
            </jdbcUpdateActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:activity name="No Records Found">
        <pd:type>com.tibco.pe.core.GenerateErrorActivity</pd:type>
        <pd:resourceType>ae.activities.throw</pd:resourceType>
        <pd:x>170</pd:x>
        <pd:y>234</pd:y>
        <config>
            <faultName>Schema0</faultName>
        </config>
        <pd:inputBindings>
            <pfx:ActivityInput>
                <message>
                    <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/DB/Messages/DBRecordNotFound"/>
                </message>
                <messageCode>
                    <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/DB/Codes/SQLRecordNotFound"/>
                </messageCode>
                <data>
                    <pfx5:DBRecordNotFound>
                        <pfx5:Message>
                            <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/DB/Messages/DBRecordNotFound"/>
                        </pfx5:Message>
                        <xsl:if test="$_globalVariables/ns1:GlobalVariables/Exceptions/DB/Codes/SQLRecordNotFound">
                            <pfx5:Code>
                                <xsl:value-of select="$_globalVariables/ns1:GlobalVariables/Exceptions/DB/Codes/SQLRecordNotFound"/>
                            </pfx5:Code>
                        </xsl:if>
                    </pfx5:DBRecordNotFound>
                </data>
            </pfx:ActivityInput>
        </pd:inputBindings>
    </pd:activity>
    <pd:transition>
        <pd:from>Start</pd:from>
        <pd:to>Update Exception Category</pd:to>
        <pd:lineType>Straight</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Update Exception Category</pd:from>
        <pd:to>End</pd:to>
        <pd:lineType>Straight</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>always</pd:conditionType>
    </pd:transition>
    <pd:transition>
        <pd:from>Update Exception Category</pd:from>
        <pd:to>No Records Found</pd:to>
        <pd:xpathDescription>No Updates</pd:xpathDescription>
        <pd:lineType>Default</pd:lineType>
        <pd:lineColor>-16777216</pd:lineColor>
        <pd:conditionType>xpath</pd:conditionType>
        <pd:xpath>$Update-Exception-Category/jdbcUpdateActivityOutput/noOfUpdates = 0</pd:xpath>
    </pd:transition>
</pd:ProcessDefinition>